migrations/2024-11-26-115042_add_combined_tables/up.sql

```diff
@@ -0,0 +1,43 @@
+-- Creates combined tables for the following:
+--
+-- Reports: (comment, post, and private_message)
+-- Inbox: (Comment replies, post replies, comment mentions, post mentions, private messages)
+-- Profile: (Posts and Comments)
+-- Modlog: (lots of types)
+-- Search: (community, post, comment, user, url)
+-- TODO not sure about these two:
+-- Home: (comment, post)
+-- Community: (comment, post)
+CREATE TABLE report_combined (
+    id serial PRIMARY KEY,
+    published timestamptz NOT NULL,
+    post_report_id int REFERENCES post_report ON UPDATE CASCADE ON DELETE CASCADE,
+    comment_report_id int REFERENCES comment_report ON UPDATE CASCADE ON DELETE CASCADE,
+    private_message_report_id int REFERENCES private_message_report ON UPDATE CASCADE ON DELETE CASCADE,
+    UNIQUE (post_report_id, comment_report_id, private_message_report_id)
```

Comment:
With CHECK it should be possible to add a constraint that exactly one of post_report_id or comment_report_id must be not null.

https://www.postgresql.org/docs/current/ddl-constraints.html